crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -413,6 +417,21 @@ fn handle_trailing_body_comment<'a>(
     // the indent-level doesn't depend on the tab width (the indent level must be the same if the tab width is 1 or 8).
     let comment_indentation_len = comment_indentation.len();
 
+    // Keep the comment on the if it's a trailing comment
```

Comment:
Nit: Maybe phrase it independently of `if` because this logic applies to all nodes